src/openrct2-ui/windows/ServerStart.cpp

```diff
@@ -47,291 +47,274 @@ enum {
+                window_start_textbox(this, widgetIndex, STR_STRING, _description, MAX_SERVER_DESCRIPTION_LENGTH);
+                break;
+            case WIDX_GREETING_INPUT:
+                window_start_textbox(this, widgetIndex, STR_STRING, _greeting, CHAT_INPUT_SIZE);
+                break;
+            case WIDX_PASSWORD_INPUT:
+                window_start_textbox(this, widgetIndex, STR_STRING, _password, 32);
+                break;
+            case WIDX_MAXPLAYERS_INCREASE:
+                if (gConfigNetwork.maxplayers < 255)
+                {
+                    gConfigNetwork.maxplayers++;
+                }
+                config_save_default();
+                OnPrepareDraw();
+                break;
+            case WIDX_MAXPLAYERS_DECREASE:
+                if (gConfigNetwork.maxplayers > 1)
+                {
+                    gConfigNetwork.maxplayers--;
+                }
+                config_save_default();
+                OnPrepareDraw();
+                break;
+            case WIDX_ADVERTISE_CHECKBOX:
+                gConfigNetwork.advertise = !gConfigNetwork.advertise;
+                config_save_default();
+                OnPrepareDraw();
+                break;
+            case WIDX_START_SERVER:
+                network_set_password(_password);
+                WindowScenarioselectOpen(ScenarioSelectCallback, false);
+                break;
+            case WIDX_LOAD_SERVER:
+                network_set_password(_password);
+                auto intent = Intent(WC_LOADSAVE);
+                intent.putExtra(INTENT_EXTRA_LOADSAVE_TYPE, LOADSAVETYPE_LOAD | LOADSAVETYPE_GAME);
+                intent.putExtra(INTENT_EXTRA_CALLBACK, reinterpret_cast<void*>(LoadSaveCallback));
+                context_open_intent(&intent);
+                break;
+        }
     }
-}
-
-static void WindowServerStartUpdate(rct_window* w)
-{
-    if (gCurrentTextBox.window.classification == w->classification && gCurrentTextBox.window.number == w->number)
+    void OnUpdate() override
     {
-        window_update_textbox_caret();
-        widget_invalidate(w, WIDX_NAME_INPUT);
-        widget_invalidate(w, WIDX_DESCRIPTION_INPUT);
-        widget_invalidate(w, WIDX_GREETING_INPUT);
-        widget_invalidate(w, WIDX_PASSWORD_INPUT);
+        if (gCurrentTextBox.window.classification == classification && gCurrentTextBox.window.number == number)
+        {
+            window_update_textbox_caret();
+            widget_invalidate(this, WIDX_NAME_INPUT);
+            widget_invalidate(this, WIDX_DESCRIPTION_INPUT);
+            widget_invalidate(this, WIDX_GREETING_INPUT);
+            widget_invalidate(this, WIDX_PASSWORD_INPUT);
+        }
     }
-}
+    void OnTextInput(rct_widgetindex widgetIndex, std::string_view text) override
+    {
+        if (text.empty())
+            return;
+
+        switch (widgetIndex)
+        {
+            case WIDX_PORT_INPUT:
+                if (strcmp(_port, text.data()) == 0)
```

Comment:
You cant do a strcmp on an string_view.data() as it might walk off the end of the array. On L182 just create a std::string of the string_view and then you can use its `.c_str()` method to compare with.